### PR TITLE
mam/api: export prefix to be defined via target 'defines' list

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: mam/defs.h [hunk 1]
```diff
@@ -24,6 +24,13 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if defined(MAM_EXPORT_JS)
```
Comment (oopsmonk) [inline mam/defs.h]: use the preprocessor define.
```suggestion
#ifdef __EMSCRIPTEN__
```

File: mam/api/BUILD [hunk 2]
```diff
@@ -31,3 +39,18 @@ map_generate(
...
+# Build command:
+# bazel clean (recommended)
+# bazel build --spawn_strategy=local --define=workspace=$(bazel info workspace)  --define=output_base=$(bazel info output_base)  --config=emscripten //mam:mam.js
+
+emcc_binary(
+    name = "api.js",
+    srcs = [
+        "api.c",
+    ],
+    defines = ["MAM_EXPORT_JS"],
```
Comment (oopsmonk) [inline mam/api/BUILD]: remove

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: mam/api/BUILD (hunk 2 above)
Comment (thibault-martinez) [reply mam/api/BUILD]: @tsvisabo Maybe that solve the issue of needing to have it under /mam/api ?

Comment (tsvisabo) [reply mam/api/BUILD]: but how then where would you define `__EMSCRIPTEN__`?

Comment (thibault-martinez) [reply mam/api/BUILD]: I believe @oopsmonk suggested it's defined by emscripten compilers, right ?

File: mam/api/api.c [hunk 3]
```diff
@@ -261,7 +261,7 @@ static retcode_t mam_api_bundle_read_packet_from_msg(mam_msg_read_context_t *ctx
  * Public functions
  */
 
-retcode_t mam_api_init(mam_api_t *const api, tryte_t const *const mam_seed) {
+MAM_EXPORT retcode_t mam_api_init(mam_api_t *const api, tryte_t const *const mam_seed) {
```
Comment (thibault-martinez) [inline mam/api/api.c]: Remove from C file

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/api/BUILD (hunk 2 above)
Comment (tsvisabo) [reply mam/api/BUILD]: could be, i'll check as soon as my pc is back to normal

Comment (oopsmonk) [reply mam/api/BUILD]: it's predefined in the toolchain.

**Review by thibault-martinez — DISMISSED** [review]

File: mam/api/BUILD (hunk 2 above)
Comment (oopsmonk) [reply mam/api/BUILD]: normally, compilers will define some flags or macros for platforms and itself that can help to distinguish the environment. we should use it, not define a new one by us. here `__EMSCRIPTEN__` is always be set when compiling app via emscripten. not sure if it's mentioned in the doc tho. :-P

**Review by oopsmonk — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**